server/api/api/utils.py

```diff
@@ -773,6 +774,61 @@ def ensure_function_has_auth_set(
             function.set_env_from_secret(env_key, secret_name, secret_key)
 
 
+def create_secret_env_vars_for_pipeline(
+    auth_info: mlrun.common.schemas.AuthInfo,
+) -> list[V1EnvVar]:
+    """
+    :param auth_info:   The auth info of the request.
+    """
+
+    if not auth_info.access_key:
+        auth_info.access_key = (
+            server.api.utils.auth.verifier.AuthVerifier().get_or_create_access_key(
+                session=auth_info.get_key(),
+                planes=[
+                    server.api.utils.clients.iguazio.SessionPlanes.data,
+                ],
+            )
+        )
+
+    mlrun_auth_secret_name = server.api.crud.Secrets().store_auth_secret(
+        secret=mlrun.common.schemas.AuthSecretData(
+            provider=mlrun.common.schemas.SecretProviderName.kubernetes,
+            username=auth_info.username,
+            access_key=auth_info.get_key(),
+        )
+    )
+    env_vars = [
+        V1EnvVar(
+            name="MLRUN_AUTH_SESSION",
+            value_from=V1EnvVarSource(
+                secret_key_ref=V1SecretKeySelector(
+                    name=mlrun_auth_secret_name,
+                    key=mlrun.common.schemas.AuthSecretData.get_field_secret_key(
+                        "access_key"
+                    ),
+                )
+            ),
+        ),
+        V1EnvVar(
+            name="V3IO_USER",
+            value=auth_info.username,
+        ),
+        V1EnvVar(
+            name="V3IO_ACCESS_KEY",
+            value_from=V1EnvVarSource(
+                secret_key_ref=V1SecretKeySelector(
+                    name=mlrun_auth_secret_name,
+                    key=mlrun.common.schemas.AuthSecretData.get_field_secret_key(
+                        "access_key"
+                    ),
+                )
+            ),
+        ),
+    ]
+    return env_vars
```

Comment:
so we do need both v3io_access_key *and* mlrun_auth_session?

Reply:
Auth session is control-plane. This also relates to my previous question - if we need both, then just a `data` plane access key will not suffice.